ouroboros-consensus/src/Ouroboros/Consensus/Util/ResourceRegistry.hs

```diff
@@ -1037,6 +1073,23 @@ releaseAllHelper rr context releaser = mask_ $ do
       Left closed -> throwRegistryClosed rr context closed
       Right keys  -> releaseResources rr keys releaser
 
+-- | Convenience combinator for allocating a resource and then doing
+-- `replaceResource`.
+replaceInRegistry :: IOLike m => ResourceKey m -> m a -> (a -> m ()) -> m a
+replaceInRegistry (ResourceKey rr rid) opener closer = do
+  context <- captureContext
+  ensureKnownThread rr context
+  a <- opener
```

Comment:
If we are to merge the PR as-is, I think we need to mask exceptions so that the opened thing always ends up in the registry.

Reply:
Outdated, closing